source/command-logging-and-monitoring/tests/monitoring/writeConcernError.yml

```diff
@@ -0,0 +1,80 @@
+description: "writeConcernError"
+schemaVersion: "1.13"
+runOnRequirements:
+  -
+    minServerVersion: 4.1.0
+    topologies:
+      - replicaset
+    serverless: "forbid"
+
+createEntities:
+  - client:
+      id: &client client
+      observeEvents:
+        - commandStartedEvent
+        - commandSucceededEvent
+        - commandFailedEvent
+  - database:
+      id: &database database
+      client: *client
+      databaseName: &databaseName command-monitoring-tests
+  - collection:
+      id: &collection collection
+      database: *database
+      collectionName: &collectionName test
+
+initialData:
+  - collectionName: *collectionName
+    databaseName: *databaseName
+    documents:
+      - { _id: 1, x: 11 }
+
+tests:
+  - description: "A retryable write with write concern errors publishes success event"
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *client
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [ insert ]
+              writeConcernError:
+                code: 91 # ShutdownInProgress
+                errorLabels: [RetryableWriteError]
+      - name: insertOne
+        object: *collection
+        arguments:
+          document: { _id: 2, x: 22 }
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collectionName
+                documents:
+                  - { _id: 2, x: 22 }
+                ordered: true
+              commandName: insert
+              databaseName: *databaseName
+          - commandSucceededEvent:
```

Comment:
the test passes in Rust; for some reason I thought we had the bug mentioned in the drivers ticket but I guess not.

that said, I did have to update some Rust logic that expected a write concern error to always have an errmsg to get this test to pass. I checked with a friend on server and she said errmsg is technically an optional field so I filed a ticket about that for Rust: https://jira.mongodb.org/browse/RUST-1574